File: Java/Data Structures/Trees/BinaryTrees/Binary-Tree-Inorder-Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    ArrayList<Integer>list;
    
    public Solution(){
        this.list = new ArrayList<>();
    }
    public List<Integer> inorderTraversal(TreeNode root) {
        if (root==null){
            return this.list;
        }
        this.inorderTraversal(root.left);
        if(root!=null){
            this.list.add(root.val);
        }
        this.inorderTraversal(root.right);
        return this.list;
    }
}

File: Leetcode Learning/Data Structures/Trees/Binary Tree/Traverse a Tree/Binary-Tree-Inorder-Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.items = []
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        if not root:
            return self.items
        self.inorderTraversal(root.left)
        if root:
            self.items.append(root.val)
        self.inorderTraversal(root.right)
        return self.items
            
